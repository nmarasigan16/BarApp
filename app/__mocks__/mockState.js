export default mockState = {
    username: 'keshavsaharia',
    profile: {
        name: 'Keshav',
        followers: 10,
        following: 15,
        public_repos: 4,
        avatar_url: 'blahblah.com',
        website: 'moomoo.com',
        email: 'crapito@gmail.com',
        bio: 'I eat cheese',
        login: 'keshavsaharia'
    },
    followStatus: 1,
    repos: [
        {
            name: 'cat',
            description: 'meow',
            owner: {
                login: 'hello'
            },
            id: 40
        },
    ],
    followers: [
        {
            name: 'geordie',
            avatar_url: 'google.com',
            id: 3,
        },
    ],
    following: [
        {
            name: 'coomeron',
            avatar_url: 'jimmyjohns.com',
            id: 5,
        },
    ],
    starredRepos: [
        {
            name: '',
            description: '',
            owner: {
                login: ''
            },
            id: 1
        }
    ]
}