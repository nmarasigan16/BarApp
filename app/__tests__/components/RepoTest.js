import 'react-native';
import React from 'react';
import Repo from '../../src/components/repos/RepoRowComponent';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import mockState from "../../__mocks__/mockState";

jest.mock('../../src/containers/repos/RepoStar', () => 'RepoStar');
it('renders correctly', () => {
    const tree = renderer.create(
        <Repo
            repo={mockState.repos[0]}/>
    ).toJSON();

    expect(tree).toMatchSnapshot();
});