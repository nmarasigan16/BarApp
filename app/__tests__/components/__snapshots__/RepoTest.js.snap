// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "flexDirection": "row",
        "height": 133.4,
        "justifyContent": "space-between",
        "marginTop": 5,
        "width": 675,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "flex": 3,
            "overflow": "hidden",
            "width": 562.5,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": "Arial",
              },
              Array [
                Object {
                  "color": "blue",
                  "fontSize": 45.911331934501746,
                  "fontWeight": "bold",
                  "marginBottom": 5.336,
                  "top": 6.67,
                },
              ],
            ]
          }
        >
          cat
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": "Arial",
              },
              Array [
                Object {
                  "fontSize": 22.955665967250873,
                  "fontStyle": "italic",
                },
              ],
            ]
          }
        >
          hello
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": "Arial",
              },
              Array [
                Object {
                  "fontSize": 19.894910504950758,
                },
              ],
            ]
          }
        >
          meow
        </Text>
      </View>
    </View>
    <RepoStar
      repo={
        Object {
          "description": "meow",
          "id": 40,
          "name": "cat",
          "owner": Object {
            "login": "hello",
          },
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#D3D3D3",
          "height": 1,
        },
        Object {
          "alignSelf": "center",
          "width": 675,
        },
      ]
    }
  />
</View>
`;
